### PR TITLE
update help command messages

### LINKED ISSUE #138: Update help command to include new commands
Add `required`, `gem`, `science`, `search`, and `tag` commands into `help`.

Update `edit`, `delete` and `find` command formats.

### PR DESCRIPTION
**Pull request by yan-soon**
Updated `help` command to include all the commands (required, gem, science, search, tags).

Also edited certain commands to keep up to date with our command format.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/HelpCommand.java [hunk 1]
```diff
@@ -6,15 +6,22 @@
...
     public static final String HELP_COMMAND = "help : Displays the help guide.";
-    public static final String LIST_COMMAND = "list : Displays list of current modules stored";
-    public static final String EDIT_COMMAND = "edit 1 c/CS1231 cr/4 :"
-            + "Replaces the current module at index 1 with module 'CS1231' with 4 credits.";
-    public static final String FIND_COMMAND = "find CS1231 : Searches for the 'CS1231' module.";
-    public static final String DELETE_COMMAND = "delete 1 : Deletes the module at index 1.";
+    public static final String LIST_COMMAND = "list : Displays list of current modules stored.";
+    public static final String EDIT_COMMAND = "edit CS1231 c/CS1101S :"
+            + "Replaces the module 'CS1231' with module 'CS1101S'.";
+    public static final String FIND_COMMAND = "find CS1 : Displays all modules that contain the prefix 'CS1'. \n"
```
Comment (Silvernitro) [inline src/main/java/seedu/address/logic/commands/HelpCommand.java]: Actually we don't do prefix search 😁  e.g. searching "103" will return "CS2103T" also
```suggestion
    public static final String FIND_COMMAND = "find CS1 : Displays all modules that contain the keyword 'CS1'. \n"
```

File: src/main/java/seedu/address/logic/commands/HelpCommand.java [hunk 2]
```diff
@@ -6,15 +6,22 @@
...
+            + "find core : Displays all modules that contain the 'core' tag.";
+    public static final String DELETE_COMMAND = "delete CS1231 : Deletes the 'CS1231' module from your GradPad.";
     public static final String CHECKMC_COMMAND = "checkmc : Checks the current total credits achieved by all modules.";
+    public static final String REQUIRED_COMMAND = "required : Displays left over required modules that you have yet"
+            + " to clear.";
+    public static final String GEM_COMMAND = "gem : Displays all available General Education modules by the Semester.";
+    public static final String SCIENCE_COMMAND = "science : Displays all available Science modules.";
+    public static final String SEARCH_COMMAND = "search MA1521 : Searches and displays all relevant information about"
```
Comment (Silvernitro) [inline src/main/java/seedu/address/logic/commands/HelpCommand.java]: Should we boast and say "from NUSMods" ah LOL

**Review by Silvernitro — COMMENTED** [review]
Just two suggestions!

File: src/main/java/seedu/address/logic/commands/HelpCommand.java (hunk 1 above)
Comment (yan-soon) [reply src/main/java/seedu/address/logic/commands/HelpCommand.java]: okay I think i seriously mixed up the definion of prefix HAHAHA

File: src/main/java/seedu/address/logic/commands/HelpCommand.java (hunk 2 above)
Comment (yan-soon) [reply src/main/java/seedu/address/logic/commands/HelpCommand.java]: MIGHT AS WELL TBH

**Review by Silvernitro — APPROVED** [review]

**Pull request closed without merging**